lib/pbench/server/indexing_tarballs.py

```diff
@@ -374,6 +375,8 @@ def sighup_handler(*args):
                                     dataset,
                                     f"Unable to find dataset: {e!s}",
                                 )
+                                if lock:
+                                    lock.release()
```

Comment:
I assume that you've already noted this, but I think you removed this change and replaced it with a context manager which produced a nice result...hopefully you can restore that code.

Reply:
Yeah; again, nothing was dropped, but the rebase restored some explicit `lock` references that are no longer relevant. The code ran just fine, though, because it also restored the leading `lock = None`. It was wrong ... but at least it was consistent.